### PR TITLE
Add check to ensure site is using WordPress 4.8

### PR DESCRIPTION
**Pull request by aaronjorbin**
Fixes https://github.com/WordPress/gutenberg/issues/1290

### CODE REVIEW

File: gutenberg.php [hunk 1]
```diff
@@ -13,11 +13,26 @@
...
+// Add check to ensure WordPress Verion.
+if ( version_compare( $wp_version, '4.8', '<' ) ) {
+	add_action( 'admin_notices', 'gutenberg_wordpress_version_notice' );
+	/**
+	 * Display a notice and deactivate Gutenberg
+	 */
+	function gutenberg_wordpress_version_notice() {
+		echo '<div class="error"><p>';
+		echo __( 'Gutenburg requires WordPress 4.8 to function properly. Please upgrade WordPress before activating Gutenburg.', 'gutenberg' );
+		echo '</p></div>';;
```
Comment (westonruter) [inline gutenberg.php]: There's a doubled semicolon here.

Comment (aaronjorbin) [inline gutenberg.php]: Thanks.

File: gutenberg.php [hunk 2]
```diff
@@ -13,11 +13,26 @@
...
+	 * Display a notice and deactivate Gutenberg
+	 */
+	function gutenberg_wordpress_version_notice() {
+		echo '<div class="error"><p>';
+		echo __( 'Gutenburg requires WordPress 4.8 to function properly. Please upgrade WordPress before activating Gutenburg.', 'gutenberg' );
+		echo '</p></div>';
 
-// Register server-side code for individual blocks.
-require_once dirname( __FILE__ ) . '/lib/blocks/latest-posts.php';
+		deactivate_plugins( array( 'gutenberg/gutenberg.php' ) );
```
Comment (nylen) [inline gutenberg.php]: I would not have thought to deactivate the plugin here.  Is this common practice, or would it be better to keep this notice around?

**Comment by nylen** [comment]
```php
deactivate_plugins( array( 'gutenberg/gutenberg.php' ) );
```

I would not have thought to deactivate the plugin here.  Is this common practice, or would it be better to keep this notice around?

**Comment by aaronjorbin** [comment]
@nylen I think it's best to deactivate the plugin so that it's very clear that Gutenberg is not in use and then after WordPress has been updated, the user can activate Gutenberg and see it in all its might.

**Comment by nylen** [comment]
A few fixes in the latest commits, including moving the PHP version check to a separate file, with the goal of keeping the `gutenberg.php` entry point extremely simple and short.

Moving the version check to its own function also fixes the build, because the plugin was [loaded via a function](https://github.com/WordPress/gutenberg/blob/5eceb502b2cef04ebb74f95cbc1e6bda4fb466d0/phpunit/bootstrap.php#L34) during PHPUnit runs, which meant that the `$wp_version` global was not available at the top level of the code.

**Comment by nylen** [comment]
Thanks for working on this.  It's working well (tested by temporarily changing `$wp_version` in my dev install); let's merge it.

**Pull request merged**